src/functions.rs

```diff
@@ -176,6 +176,60 @@ pub(crate) fn render_servers(app: &Rc<App>) {
                 app_clone.connections.set_current(Some(addr));
                 app_clone.message_edit.set_reveal_child(false);
                 render_channels(Some(addr), &app_clone);
+
+                app_clone.connections.execute(addr, |result| {
```

Comment:
There is another `execute` call right above here. You should be able to use that instead, to avoid calling it twice.